### PR TITLE
Issue 3836

### PR DESCRIPTION
**Pull request by fritzschoff**
ref #3836

### CODE REVIEW

File: apps/financiers/financiers/src/app/marketplace/organization/list/list.component.scss [hunk 1]
```diff
@@ -0,0 +1,13 @@
+mat-form-field {
+  width: 100%;
+}
+
+.page-title {
+  text-align: center;
+  font-size: 3rem;
```
Comment (GrandSchtroumpf) [inline apps/financiers/financiers/src/app/marketplace/organization/list/list.component.scss:7]: don't override theme typography.

File: apps/financiers/financiers/src/app/marketplace/organization/list/list.component.ts [hunk 2]
```diff
@@ -15,11 +19,21 @@ import { DynamicTitleService } from '@blockframes/utils/dynamic-title/dynamic-ti
...
   changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class ListComponent implements OnInit {
+export class ListComponent implements OnInit, OnDestroy {
 
   @HostBinding('@scaleOut') animation = true;
   orgs$: Observable<Organization[]>;
-  public movies: Movie[] = [];
+
+  private organizationSearchResultsState = new BehaviorSubject<Organization[]>([]);
```
Comment (GrandSchtroumpf) [inline apps/financiers/financiers/src/app/marketplace/organization/list/list.component.ts:27]: just "results" or "orgs" ^^

File: apps/financiers/financiers/src/app/marketplace/organization/list/list.component.ts [hunk 3]
```diff
@@ -15,11 +19,21 @@ import { DynamicTitleService } from '@blockframes/utils/dynamic-title/dynamic-ti
...
+
+  public orgSearchForm = new OrganizationSearchForm();
```
Comment (GrandSchtroumpf) [inline apps/financiers/financiers/src/app/marketplace/organization/list/list.component.ts:29]: just searchForm. We are already in the org component so we know we're searching for orgs

File: apps/financiers/financiers/src/app/marketplace/organization/list/list.component.ts [hunk 4]
```diff
@@ -28,10 +42,47 @@ export class ListComponent implements OnInit {
...
   ngOnInit() {
     this.dynTitle.setPageTitle('Sales Agent', 'All');
-    this.orgs$ = this.service
-      .valueChanges(ref => ref
-        .where('appAccess.financiers.dashboard', '==', true)
-        .where('status', '==', 'accepted'))
-      .pipe(map(orgs => orgs.filter((org: Organization) => org.id !== centralOrgID && org.movieIds.length)));
+
+    this.orgs$ = this.organizationSearchResultsState.asObservable();
+    this.orgSearchForm.setValue(createOrganizationSearch({ appAccess: ['financiers'], appModule: ['marketplace'] }))
```
Comment (GrandSchtroumpf) [inline apps/financiers/financiers/src/app/marketplace/organization/list/list.component.ts:47]: use two lines: 
```typescript
const search = createOrganizationSearch({ appAccess: ['financiers'], appModule: ['marketplace'] });
this.searchForm.setValue(search);
```

File: apps/financiers/financiers/src/app/marketplace/organization/list/list.component.ts [hunk 5]
```diff
@@ -28,10 +42,47 @@ export class ListComponent implements OnInit {
...
+      startWith(this.orgSearchForm.value),
+      distinctUntilChanged(),
+      debounceTime(500),
+      switchMap(() => this.orgSearchForm.search()),
+      tap(res => this.nbHits = res.nbHits),
+      pluck('hits'),
+      map(results => results.map(org => org.objectID)),
+      switchMap(ids => ids.length ? this.service.valueChanges(ids) : of([])),
+      // map(movies => movies.sort((a, b) => sortMovieBy(a, b, this.sortByControl.value))), // TODO issue #3584
+      tap(orgs => {
```
Comment (GrandSchtroumpf) [inline apps/financiers/financiers/src/app/marketplace/organization/list/list.component.ts:58]: use subscribe directly. 
```typescript
subscribe(orgs => {

})
```

File: libs/organization/src/lib/organization/forms/search.form.ts [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+    page: new FormControl(search.page),
+    country: new FormControl(search.country),
+    appAccess: new FormControl(search.appAccess),
+    appModule: new FormControl(search.appModule)
+  };
+}
+
+export type OrganizationSearchControl = ReturnType<typeof createOrganizationSearchControl>;
+
+export class OrganizationSearchForm extends FormEntity<OrganizationSearchControl> {
```
Comment (GrandSchtroumpf) [inline libs/organization/src/lib/organization/forms/search.form.ts:41]: Is there no way to merge some part with movie search ? could be in another PR

File: libs/ui/src/lib/list/page/list-page.component.html [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+  </a>
+  <ng-template [ngTemplateOutlet]="pageTitleTemplate"></ng-template>
+  <section fxLayout="row" fxLayoutAlign="space-between center" fxLayoutGap="32px">
+
+    <span class="search">
+      <!-- MAIN SEARCH BAR -->
+      <ng-template [ngTemplateOutlet]="pageSearchTemplate"></ng-template>
+    </span>
+
+    <span>
```
Comment (GrandSchtroumpf) [inline libs/ui/src/lib/list/page/list-page.component.html:14]: Remove this span. It doesn't do anything.

File: libs/ui/src/lib/list/page/list-page.component.html [hunk 8]
```diff
@@ -0,0 +1,78 @@
+<!-- we use class="dark-contrast-theme" to override theme style of the header because it always has a "dark" background image -->
+<header class="dark-contrast-theme" bgAsset="banner_header_page.webp">
+  <a mat-icon-button (click)="goBack()">
+    <mat-icon svgIcon="arrow_back"></mat-icon>
+  </a>
+  <ng-template [ngTemplateOutlet]="pageTitleTemplate"></ng-template>
+  <section fxLayout="row" fxLayoutAlign="space-between center" fxLayoutGap="32px">
+
+    <span class="search">
```
Comment (GrandSchtroumpf) [inline libs/ui/src/lib/list/page/list-page.component.html:9]: Shouldn't be inside a span. either don't wrap the ng-template inside anything or a div or an article

File: libs/ui/src/lib/list/page/list-page.component.html [hunk 9]
```diff
@@ -0,0 +1,78 @@
...
+<!-- FILTER SIDE CARD -->
+<div fxLayout fxLayoutAlign="space-between center" class="filter dark-contrast-theme" bgAsset="banner_header_page.webp">
+  <ng-content></ng-content>
+</div>
+
+<!-- Items -->
+<section>
+  <ng-container *ngIf="items.length; else error">
+    <div fxLayout="row" fxLayoutAlign="end start">
+      <ng-template [ngTemplateOutlet]="pageResultDescriptionTemplate"></ng-template>
```
Comment (GrandSchtroumpf) [inline libs/ui/src/lib/list/page/list-page.component.html:40]: pageResultDescriptionTemplate -> descriptionTpl

File: libs/ui/src/lib/list/page/list-page.component.html [hunk 10]
```diff
@@ -0,0 +1,78 @@
...
+
+      </div>
+    </ng-template>
+
+    <!-- PROGRESS -->
+    <ng-template [ngTemplateOutlet]="pageProgressTemplate"></ng-template>
+  </ng-container>
+
+  <!-- ERROR HANDLING -->
+  <ng-template #error>
```
Comment (GrandSchtroumpf) [inline libs/ui/src/lib/list/page/list-page.component.html:69]: Maybe #empty makes more sense.

File: libs/ui/src/lib/list/page/list-page.component.html (hunk 9 above)
Comment (GrandSchtroumpf) [reply libs/ui/src/lib/list/page/list-page.component.html:40]: Also I think you should use ng-content also here.

File: libs/ui/src/lib/list/page/list-page.component.html [hunk 11]
```diff
@@ -0,0 +1,78 @@
...
+    <div fxLayout="row" fxLayoutAlign="end start">
+      <ng-template [ngTemplateOutlet]="pageResultDescriptionTemplate"></ng-template>
+    </div>
+    <ng-container *ngIf="!listView; else displayList">
+      <div grid [@fadeList]="items.length">
+        <ng-container *ngFor="let item of items; trackBy:trackById">
+          <div [col]="3">
+
+            <!-- INSERT CARD -->
+            <ng-template [ngTemplateOutlet]="pageCardTemplate" [ngTemplateOutletContext]="{$implicit: item}">
```
Comment (GrandSchtroumpf) [inline libs/ui/src/lib/list/page/list-page.component.html:48]: pageCardTemplate -> cardTpl

File: libs/ui/src/lib/list/page/list-page.component.html [hunk 12]
```diff
@@ -0,0 +1,78 @@
...
+            </ng-template>
+          </div>
+        </ng-container>
+      </div>
+    </ng-container>
+    <ng-template #displayList>
+      <div *ngFor="let item of items">
+
+        <!-- INSERT LIST ITEM -->
+        <ng-template [ngTemplateOutlet]="pageListItemTemplate" [ngTemplateOutletContext]="{$implicit: item}">
```
Comment (GrandSchtroumpf) [inline libs/ui/src/lib/list/page/list-page.component.html:58]: pageListItemTemplate -> listItemTpl

File: libs/ui/src/lib/list/page/list-page.component.scss [hunk 13]
```diff
@@ -0,0 +1,39 @@
...
+  width: 100%;
+
+  section {
+    padding: 20px 20px 0;
+    .search {
+      width: 100%;
+      margin-top: 24px;
+    }
+  }
+  mat-icon {
```
Comment (GrandSchtroumpf) [inline libs/ui/src/lib/list/page/list-page.component.scss:15]: You don't have mat-icon. Also a mat-icon with pointer should be inside a mat-icon-button

File: libs/ui/src/lib/list/page/list-page.component.scss [hunk 14]
```diff
@@ -0,0 +1,39 @@
...
+    cursor: pointer;
+  }
+}
+
+.filter {
+  position: sticky;
+  top: 0;
+  z-index: 2;
```
Comment (GrandSchtroumpf) [inline libs/ui/src/lib/list/page/list-page.component.scss:23]: Couldn't be 1 ?

File: libs/ui/src/lib/load-more-button/load-more-button.component.html [hunk 15]
```diff
@@ -0,0 +1,6 @@
+<div>
```
Comment (GrandSchtroumpf) [inline libs/ui/src/lib/load-more-button/load-more-button.component.html:1]: Remove the div. Style the :host instead

File: libs/ui/src/lib/load-more-button/load-more-button.component.ts [hunk 16]
```diff
@@ -0,0 +1,29 @@
...
+  Component,
+  ContentChild,
+  Directive,
+  EventEmitter,
+  Input,
+  Output,
+  TemplateRef
+} from '@angular/core';
+
+@Directive({ selector: '[buttonText]' })
```
Comment (GrandSchtroumpf) [inline libs/ui/src/lib/load-more-button/load-more-button.component.ts:13]: You should use ng-content instead.

**Review by GrandSchtroumpf — CHANGES_REQUESTED** [review]

File: libs/ui/src/lib/load-more-button/load-more-button.component.html (hunk 15 above)
Comment (fritzschoff) [reply libs/ui/src/lib/load-more-button/load-more-button.component.html:1]: If I would remove that puppy, the progressbar will be the whole width... I know, not the best solution

Comment (GrandSchtroumpf) [reply libs/ui/src/lib/load-more-button/load-more-button.component.html:1]: set the width of the process to 80% ?

File: apps/financiers/financiers/src/app/marketplace/organization/list/list.component.html [hunk 17]
```diff
@@ -1,8 +1,41 @@
-<section grid>
-  <ng-container *ngIf="(orgs$ | async) as orgs; else loading">
-    <org-card *ngFor="let org of orgs" [col]="3" [org]="org"></org-card>
-  </ng-container>
-</section>
-<ng-template #loading>
-  <mat-progress-spinner mode="indeterminate"></mat-progress-spinner>
-</ng-template>
+<list-page [items]="orgs$ | async">
+  <page-title>
```
Comment (GrandSchtroumpf) [inline apps/financiers/financiers/src/app/marketplace/organization/list/list.component.html:2]: use `list-page-title` to match angular recommendation on selector naming

File: apps/financiers/financiers/src/app/marketplace/organization/list/list.component.html [hunk 18]
```diff
@@ -1,8 +1,41 @@
...
+  </page-search>
+
+  <page-description>
+    <span i18n *ngIf="nbHits" class="mat-body-2">
+      There {nbHits, plural, =0 {} =1 {is} other {are}} {{ nbHits }} partners{nbHits, plural, =0 {} =1 {} other {s}}
+      available.
+    </span>
+  </page-description>
+
+  <list-filter>
```
Comment (GrandSchtroumpf) [inline apps/financiers/financiers/src/app/marketplace/organization/list/list.component.html:20]: **if** this directive is supposed to be used **only** in the list-page, you can name it "list-page-filter" for coherence with the rest

File: apps/financiers/financiers/src/app/marketplace/organization/list/list.component.html [hunk 19]
```diff
@@ -1,8 +1,41 @@
...
+    <ng-template filter label="Partner Country" [form]="searchForm.country">
+      <form-country [form]="searchForm.country"></form-country>
+    </ng-template>
+  </list-filter>
+
+  <ng-template pageCard let-org>
+    <org-card [org]="org"></org-card>
+  </ng-template>
+
+  <page-empty *ngIf="loading$.asObservable() | async">
```
Comment (GrandSchtroumpf) [inline apps/financiers/financiers/src/app/marketplace/organization/list/list.component.html:30]: you don't need to set `asObservable`. The `asObservable` only delete the `.next` function. In this context it doesn't help.

File: libs/ui/src/lib/load-more-button/load-more-button.component.html (hunk 15 above)
Comment (fritzschoff) [reply libs/ui/src/lib/load-more-button/load-more-button.component.html:1]: I guess then you get a big margin right, cause the parent element doesn't have flex

File: apps/financiers/financiers/src/app/marketplace/organization/list/list.component.html [hunk 20]
```diff
@@ -1,8 +1,41 @@
...
+    <div class="loading">
+      <mat-spinner color="primary"></mat-spinner>
+    </div>
+  </page-empty>
+
+  <page-progress>
```
Comment (GrandSchtroumpf) [inline apps/financiers/financiers/src/app/marketplace/organization/list/list.component.html:36]: As a reader I wonder why not doing : 
```html
<page-progress [value]="hitsViewed / nbHits * 100 | percentage" (loadMore)="loadMore()">
  <span>You've viewed {{ hitsViewed }} partners from {{ nbHits }}.</span>
</page-progress>
```

File: libs/ui/src/lib/list/page/list-page.component.html [hunk 21]
```diff
@@ -0,0 +1,77 @@
...
+
+    <!-- SORT CONTROL-->
+    <ng-content select="page-sort"></ng-content>
+
+  </section>
+</header>
+
+<!-- FILTER SIDE CARD -->
+<div fxLayout fxLayoutAlign="space-between center" class="filter dark-contrast-theme" bgAsset="banner_header_page.webp">
+  <ng-content></ng-content>
```
Comment (GrandSchtroumpf) [inline libs/ui/src/lib/list/page/list-page.component.html:31]: Don't you need a select on this ng-content ? Like `ng-content select="list-filter"`

File: libs/ui/src/lib/list/page/list-page.component.html (hunk 11 above)
Comment (GrandSchtroumpf) [reply libs/ui/src/lib/list/page/list-page.component.html:46]: Please change to `cardTemplate` as least. You don't have to change the directive name, just the name of the property.

File: libs/ui/src/lib/list/page/list-page.component.html (hunk 12 above)
Comment (GrandSchtroumpf) [reply libs/ui/src/lib/list/page/list-page.component.html:56]: Please change it to `listItemTemplate` at least

**Review by GrandSchtroumpf — CHANGES_REQUESTED** [review]

File: libs/ui/src/lib/load-more-button/load-more-button.component.html (hunk 15 above)
Comment (GrandSchtroumpf) [reply libs/ui/src/lib/load-more-button/load-more-button.component.html:1]: Maybe this would work ?
```scss
:host {
  display: flex;
  flex-direction: column;
  align-item: center;
  justify-content: center;
  width: 100%;
}
mat-progress-bar {
  width: 80%;
}
```

File: apps/financiers/financiers/src/app/marketplace/organization/list/list.component.html (hunk 20 above)
Comment (fritzschoff) [reply apps/financiers/financiers/src/app/marketplace/organization/list/list.component.html:36]: cause the inputs are for the bf-load-more-button component and we want to reuse it

Comment (fritzschoff) [reply apps/financiers/financiers/src/app/marketplace/organization/list/list.component.html]: whops, guess my comment got lost. It is a reusable component the bf-load-more-button. Pretty sure you want to show something else here in the future

**Review by GrandSchtroumpf — APPROVED** [review]
